llvm/lib/Transforms/Scalar/GVN.cpp

```diff
@@ -833,6 +839,8 @@ PreservedAnalyses GVNPass::run(Function &F, FunctionAnalysisManager &AM) {
       isMemDepEnabled() ? &AM.getResult<MemoryDependenceAnalysis>(F) : nullptr;
   auto &LI = AM.getResult<LoopAnalysis>(F);
   auto *MSSA = AM.getCachedResult<MemorySSAAnalysis>(F);
+  if (isMemorySSAEnabled() && !MSSA)
+    MSSA = &AM.getResult<MemorySSAAnalysis>(F);
```

Comment:
Can you add the assert inside this if clause (`assert (!MemDep && "...");`)?
If MSSA is already available, it's ok to use. But if it's unavailable and we pay the cost of computing from scratch, we should be able to drop the use MemDep. This shouldn't impact the compiler tracker, but it should make the pass crash if trying to turn on before implementing the switch.

Reply:
That makes sense, updated, thanks!